File: src/main/java/com/GoldenMine/parser/predicatespecific/IPredicateSpecific.java
package com.GoldenMine.parser.predicatespecific;

import com.GoldenMine.parser.Sentence;
import com.GoldenMine.parser.Variable;
import com.GoldenMine.parser.VariableStorage;
import java.util.ArrayList;
import java.util.List;
import kr.co.shineware.nlp.komoran.model.KomoranResult;

public interface IPredicateSpecific {
    boolean verify(String source, KomoranResult result);

    Variable perform(Sentence sentence, VariableStorage local, Variable originalResult);

    List<String> getOthers();

    int execute(int line, Sentence sentence, MultiProcessData parameters, Variable result);

    class MultiProcessData {
        private List<Integer> applyLine;
        private Object meta;

        public MultiProcessData(List<Integer> applyLine) {
            this.applyLine = applyLine;
        }

        public int getAppliedValue(int index) {
            return applyLine.get(index);
        }

        public int getSize() {
            return applyLine.size();
        }

        public Object getMeta() {
            return meta;
        }

        public void setMeta(Object meta) {
            this.meta = meta;
        }
    }
}

File: src/main/java/kr/goldenmine/parser/predicatespecific/IPredicateSpecific.kt
package kr.goldenmine.parser.predicatespecific

import kr.goldenmine.parser.Sentence
import kr.goldenmine.parser.Variable
import kr.goldenmine.parser.VariableStorage
import kr.co.shineware.nlp.komoran.model.KomoranResult

interface IPredicateSpecific {

    val others: List<String>
    fun verify(source: String, result: KomoranResult): Boolean

    fun perform(sentence: Sentence, metadata: List<Any>?, local: VariableStorage, originalResult: MutableList<Variable?>?): Variable?

    fun execute(line: Int, metadata: List<Any>?, sentence: Sentence, parameters: MultiProcessData?, results: List<Variable?>?): Int

    class MultiProcessData(private val applyLine: List<Int>) {
        var meta: Any? = null

        val size: Int
            get() = applyLine.size

        fun getAppliedValue(index: Int): Int {
            return applyLine[index]
        }

        override fun toString(): String {
            return "$applyLine, $meta"
        }
    }
}
